APE10.rst

```diff
@@ -0,0 +1,178 @@
+  updating all the code that used ``astropy.extern.six`` to use the Python 3
+  syntax directly.
+* Removing Python 2.7 from the configuration for the continuous integration
+  services (such as Travis and AppVeyor at the time of writing)
+* Making sure that the documentation on ReadTheDocs is built using Python 3.x
+* Removing any backports that may exist for Python 2.7 compatibility
+* Updating the documentation to remove any mention of Python 2.7, as well as
+  making sure that Python 3.x is not presented as a special case.
+* Removing Python 2 and 2.7 from the PyPI classifiers
+
+Challenges
+----------
+
+There are two main challenges with the plan outlined above:
+
+Developers will need to use Python 3
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once we make the switch to a Python 3-only code base, between v2.0 and v3.0,
+people will no longer be able to contribute code to Astropy without using
+Python 3. This means that even though users can in theory wait until the end of
+2019 to switch to Python 3, developers and contributors will need to make the
+switch (at least for their Astropy development environment) as soon as 2017.
+
+Packages on PyPI
+^^^^^^^^^^^^^^^^^
+
+If we upload Astropy v3.0 to PyPI, and it does not support Python 2, then if
+Python 2.7 users try and pip install Astropy, the install or import will fail.
+This is because PyPI does not compare the current Python version to the PyPI
+meta-data for the package (which may indicate for example whether the package
+is Python 2-compatible), and will download Astropy v3.0 regardless of the
+active Python version. However, there is a solution, which is described in the 
+`Jupyter roadmap <https://github.com/jupyter/roadmap/blob/master/accepted/migration-to-python-3-only.md#multiple-source-distributions>`_.
+Essentially, the solution for Python 3-only releases is to upload tar files
+that have a ``-py3.x`` suffix (one file per Python 3.x version), or hopefully,
+if this is fixed in `pip <https://pip.pypa.io/en/stable/>`_, a single file with a ``-py3`` suffix would be
+sufficient (e.g. ``astropy-3.0-py3.5.tar.gz``).
+
+Benefits
+--------
+
+There are several benefits to following the plan proposed above:
+
+* Since we will need to keep adding Python 3.x releases to the continuous
+  integration over the coming years, we will at least be able to remove Python
+  2.7, making sure that the number of builds does not grow out of control.
+
+* Since developers/contributors will need to switch to using Python 3 for
+  Astropy development, we will be training more people to do this transition,
+  who will then be able to help their colleagues also make the transition.
+
+* We will be able to start using Python 3-only features internally, including
+  for example function annotations (e.g., for units), matrix multiplication
+  (e.g., for coordinates; note that this will only be possible once we support
+  only Python 3.5+)
+
```

Comment:
One more advantage to add (one of the biggest in my mind!): It will make maintaining and writing the code significantly easier, because devs don't have to know both 2.x and 3.x (and `six`) and the various "gotchas" involved there.  They can just code consistently for py 3.x only.